specifyweb/frontend/js_src/lib/components/FormSliders/RecordSelectorFromIds.tsx

```diff
@@ -218,15 +227,25 @@ export function RecordSelectorFromIds<SCHEMA extends AnySchema>({
                   onClick={(): void => handleRemove('minusButton')}
                 />
               ) : undefined}
-
               {typeof newResource === 'object' ? (
                 <p className="flex-1">{formsText.creatingNewRecord()}</p>
               ) : (
                 <span
                   className={`flex-1 ${dialog === false ? '-ml-2' : '-ml-4'}`}
                 />
               )}
-
+              {hasAttachments && (
```

Comment:
Consider extracting this attachment-related logic into a separate component, just to keep this file as devoid of attachment-related logic as possible.

For example, imagine this file just has this line:
```
<RecordSetAttachments records={records} onFetch={handleFetch} />
```

And then the actual button, the state that holds whether the attachment dialog is open and the actual RecordSetAttachments dialog would be handled in that component



(`RecordSetAttachments` would be a new component, and currently named `RecordSetAttachments` component could be renamed to something like `AttachmentsDialog` and won't need to be exported anymore as it would be in same file as `RecordSetAttachments`)